drivers/gpu/drm/panel/msm8916-generated/panel-xiaomi-lavender-td4320.c

```diff
@@ -0,0 +1,247 @@
+// SPDX-License-Identifier: GPL-2.0-only
+// Copyright (c) 2024 FIXME
+// Generated with linux-mdss-dsi-panel-driver-generator from vendor device tree:
+//   Copyright (c) 2013, The Linux Foundation. All rights reserved. (FIXME)
+
+#include <linux/delay.h>
+#include <linux/gpio/consumer.h>
+#include <linux/module.h>
+#include <linux/of.h>
+
+#include <video/mipi_display.h>
+
+#include <drm/drm_mipi_dsi.h>
+#include <drm/drm_modes.h>
+#include <drm/drm_panel.h>
+
+struct td4320_boeplus {
+	struct drm_panel panel;
+	struct mipi_dsi_device *dsi;
+	struct gpio_desc *reset_gpio;
+	bool prepared;
+};
+
+static inline struct td4320_boeplus *to_td4320_boeplus(struct drm_panel *panel)
+{
+	return container_of(panel, struct td4320_boeplus, panel);
+}
+
+static void td4320_boeplus_reset(struct td4320_boeplus *ctx)
+{
+	gpiod_set_value_cansleep(ctx->reset_gpio, 0);
+	usleep_range(1000, 2000);
+	gpiod_set_value_cansleep(ctx->reset_gpio, 1);
+	usleep_range(5000, 6000);
+	gpiod_set_value_cansleep(ctx->reset_gpio, 0);
+	msleep(30);
+}
+
+static int td4320_boeplus_on(struct td4320_boeplus *ctx)
+{
+	struct mipi_dsi_device *dsi = ctx->dsi;
+	struct device *dev = &dsi->dev;
+	int ret;
+
+	dsi->mode_flags |= MIPI_DSI_MODE_LPM;
+
+	mipi_dsi_generic_write_seq(dsi, 0xb0, 0x04);
+	mipi_dsi_generic_write_seq(dsi, 0xd6, 0x00);
+	mipi_dsi_generic_write_seq(dsi, 0xb8,
+				   0x19, 0x55, 0x00, 0xbe, 0x00, 0x00, 0x00);
+	mipi_dsi_generic_write_seq(dsi, 0xb9,
+				   0x4d, 0x55, 0x05, 0xe6, 0x00, 0x02, 0x03);
+	mipi_dsi_generic_write_seq(dsi, 0xba,
+				   0x9b, 0x5b, 0x07, 0xe6, 0x00, 0x13, 0x00);
+	mipi_dsi_generic_write_seq(dsi, 0xf9, 0x44, 0x3f, 0x00, 0x8d, 0xbf);
+	mipi_dsi_generic_write_seq(dsi, 0xce,
+				   0x5d, 0x00, 0x0f, 0x1f, 0x2f, 0x3f, 0x4f,
+				   0x5f, 0x6f, 0x7f, 0x8f, 0x9f, 0xaf, 0xbf,
+				   0xcf, 0xdf, 0xef, 0xff, 0x04, 0x00, 0x02,
+				   0x02, 0x42, 0x01, 0x69, 0x5a, 0x40, 0x40,
+				   0x00, 0x00, 0x04, 0xfa, 0x00);
+
+	ret = mipi_dsi_dcs_set_display_brightness(dsi, 0x00b8);
+	if (ret < 0) {
+		dev_err(dev, "Failed to set display brightness: %d\n", ret);
+		return ret;
+	}
+
+	mipi_dsi_dcs_write_seq(dsi, MIPI_DCS_WRITE_CONTROL_DISPLAY, 0x2c);
+	mipi_dsi_dcs_write_seq(dsi, MIPI_DCS_WRITE_POWER_SAVE, 0x00);
+	mipi_dsi_dcs_write_seq(dsi, 0xb0, 0x03);
+	mipi_dsi_dcs_write_seq(dsi, 0x11, 0x00);
+	msleep(96);
+	mipi_dsi_dcs_write_seq(dsi, 0x29, 0x00);
+	msleep(20);
+
+	return 0;
+}
+
+static int td4320_boeplus_off(struct td4320_boeplus *ctx)
+{
+	struct mipi_dsi_device *dsi = ctx->dsi;
+
+	dsi->mode_flags &= ~MIPI_DSI_MODE_LPM;
+
+	mipi_dsi_dcs_write_seq(dsi, 0x28, 0x00);
+	msleep(20);
+	mipi_dsi_dcs_write_seq(dsi, 0x10, 0x00);
+	msleep(120);
+
+	return 0;
+}
+
+static int td4320_boeplus_prepare(struct drm_panel *panel)
+{
+	struct td4320_boeplus *ctx = to_td4320_boeplus(panel);
+	struct device *dev = &ctx->dsi->dev;
+	int ret;
+
+	if (ctx->prepared)
+		return 0;
+
+	td4320_boeplus_reset(ctx);
```

Comment:
I remember you told it only works with reset commented out. It it still true?

Reply:
No, the issue was i have used ```reset-gpios = <&tlmm 53 GPIO_ACTIVE_HIGH>;``` but it works fine with ```reset-gpios = <&tlmm 53 GPIO_ACTIVE_LOW>;```